pkg/environ/providers/vault/main.go

```diff
@@ -135,8 +144,41 @@ func (c *Client) AddToEnviron(e *environ.Environ) error {
 				env[k] = s
 			}
 		}
+
 		e.SafeMerge(env)
 	}
+
+	if c.IamRole != "" {
+		route := "aws/sts/" + c.IamRole
```

Comment:
Might be a good idea to make the mount customizable (the `aws` part). But looks good otherwise.

Reply:
not a bad idea